thunder/tests/test_examine_memory.py

```diff
@@ -71,7 +71,7 @@ def bar(a, b):  # [4] [2,2]
     bw_traces = thunder.last_backward_traces(cbar)
     bw_extrace = bw_traces[-1]
     max_mem_bw = get_alloc_memory(bw_extrace)
-    assert max_mem_bw[0] == 144
+    assert max_mem_bw[0] == 128
```

Comment:
How did the trace change?

Reply:
We go from this:
```python
@torch.no_grad()
@no_autocast
def backward_fn(saved_for_backward, cotangents):
  # cotangents: "Collection"
  t6, t7, = cotangents
  clear_mutable_collection(cotangents)
  del cotangents
  [t58, t69] = nvFusion0(t6, t7)
    # t64 = prims.sum(t7, (0, 1))  # t64: "cuda:0 f32[4]"
    # t65 = prims.broadcast_in_dim(t64, [1, 1, 4], [2])  # t65: "cuda:0 f32[1, 1, 4]"
    # t66 = prims.add(t6, t65)  # t66: "cuda:0 f32[1, 1, 4]"
    # t67 = prims.sum(t66, (0, 1))  # t67: "cuda:0 f32[4]"
    # t68 = prims.broadcast_in_dim(t67, [1, 4], [1])  # t68: "cuda:0 f32[1, 4]"
    # t69 = prims.sum(t68, (0,))  # t69: "cuda:0 f32[4]"
    # t54 = prims.broadcast_in_dim(t64, [1, 1, 4], [2])  # t54: "cuda:0 f32[1, 1, 4]"
    # t55 = prims.add(t6, t54)  # t55: "cuda:0 f32[1, 1, 4]"
    # t56 = prims.sum(t55, (0, 1))  # t56: "cuda:0 f32[4]"
    # t57 = prims.broadcast_in_dim(t56, [1, 4], [1])  # t57: "cuda:0 f32[1, 4]"
    # t58 = prims.sum(t57, (0,))  # t58: "cuda:0 f32[4]"
  del t6, t7
  t60 = torch.reshape(t58, (2, 2))  # t60: "cuda:0 f32[2, 2]"
    # t60 = ltorch.reshape(t58, (2, 2))  # t60: "cuda:0 f32[2, 2]"
      # t60 = prims.reshape(t58, (2, 2))  # t60: "cuda:0 f32[2, 2]"
  del t58
  return (t69, t60)
```

to this: 
```python
@torch.no_grad()
@no_autocast
def backward_fn(saved_for_backward, cotangents):
  # cotangents: "Collection"
  t6, t7, = cotangents
  clear_mutable_collection(cotangents)
  del cotangents
  [t69] = nvFusion0(t6, t7)
    # t64 = prims.sum(t7, (0, 1))  # t64: "cuda:0 f32[4]"
    # t65 = prims.broadcast_in_dim(t64, [1, 1, 4], [2])  # t65: "cuda:0 f32[1, 1, 4]"
    # t66 = prims.add(t6, t65)  # t66: "cuda:0 f32[1, 1, 4]"
    # t67 = prims.sum(t66, (0, 1))  # t67: "cuda:0 f32[4]"
    # t68 = prims.broadcast_in_dim(t67, [1, 4], [1])  # t68: "cuda:0 f32[1, 4]"
    # t69 = prims.sum(t68, (0,))  # t69: "cuda:0 f32[4]"
  del t6, t7
  t60 = torch.reshape(t69, (2, 2))  # t60: "cuda:0 f32[2, 2]"
    # t60 = ltorch.reshape(t69, (2, 2))  # t60: "cuda:0 f32[2, 2]"
      # t60 = prims.reshape(t69, (2, 2))  # t60: "cuda:0 f32[2, 2]"
  return (t69, t60)
```
Thanks @kiya00 for the discussion on this test and the traces!